modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/DuplicateEventWorkflowOperationHandler.java

```diff
@@ -309,14 +328,45 @@ public WorkflowOperationResult start(final WorkflowInstance workflowInstance, fi
     for (int i = 0; i < numberOfEvents; i++) {
       final List<URI> temporaryFiles = new ArrayList<>();
       MediaPackage newMp = null;
-
+      final Date startDate;
       try {
         String newMpId = workflowInstance.getConfiguration("newMpId");
         if (newMpId == null) {
           newMpId = UUID.randomUUID().toString();
         }
         // Clone the media package (without its elements)
-        newMp = copyMediaPackage(mediaPackage, series, newMpId, noSuffix, i + 1, copyNumberPrefix);
+        String useTitle;
+        if (title.isEmpty() || (title.startsWith("${") && title.endsWith("}"))) {
+          useTitle = mediaPackage.getTitle();
+        } else {
+          useTitle = title;
+        }
+        final String newTitle = noSuffix
+            ? useTitle
+            : String.format("%s (%s %d)", useTitle, copyNumberPrefix, i + 1);
```

Comment:
I did not go through all the code but this seems a complex way of:

```suggestion
        if (title.isEmpty()) {
          title = mediaPackage.getTitle();
        }
        if (!noSuffix) {
          title = String.format("%s (%s %d)", title, copyNumberPrefix, i + 1);
        }
```

Reply:
code simplified :-)